lib/ansible/modules/database/influxdb/influxdb_user.py

```diff
@@ -142,13 +159,51 @@ def drop_user(module, client, user_name):
     module.exit_json(changed=True)
 
 
+def set_user_grants(module, client, user_name, grants):
+    changed = False
+
+    try:
+        current_grants = client.get_list_privileges(user_name)
+        # Fix privileges wording
+        for i, v in enumerate(current_grants):
+            if v['privilege'] == 'ALL PRIVILEGES':
+                v['privilege'] = 'ALL'
+                current_grants[i] = v
+            elif v['privilege'] == 'NO PRIVILEGES':
+                del(current_grants[i])
+
+        # check if the current grants are included in the desired ones
+        for current_grant in current_grants:
+            if current_grant not in grants:
+                if not module.check_mode:
+                    client.revoke_privilege(current_grant['privilege'],
+                                            current_grant['database'],
+                                            user_name)
+                changed = True
+
+        # check if the desired grants are included in the current ones
+        for grant in grants:
+            if grant not in current_grants:
+                if not module.check_mode:
+                    client.grant_privilege(grant['privilege'],
+                                           grant['database'],
+                                           user_name)
+                changed = True
+
+    except influx.exceptions.InfluxDBClientError as e:
+        module.fail_json(msg=e.content)
+
+    return changed
+
+
 def main():
     argument_spec = influx.InfluxDb.influxdb_argument_spec()
     argument_spec.update(
         state=dict(default='present', type='str', choices=['present', 'absent']),
         user_name=dict(required=True, type='str'),
         user_password=dict(required=False, type='str', no_log=True),
-        admin=dict(default='False', type='bool')
+        admin=dict(default='False', type='bool'),
+        grants=dict(default=[], type='list')
```

Comment:
Default to `None`:
~~~python
grants=dict(type='list')
~~~

Reply:
It's done, thanks